sklearn/multioutput.py

```diff
@@ -359,3 +362,234 @@ def score(self, X, y):
+        for column 3, etc.
+
+        If order is 'random' a random ordering will be used.
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines whether to use cross validated predictions or true
+        labels for the results of previous estimators in the chain.
+        If cv is None the true labels are used when fitting. Otherwise
+        possible inputs for cv are:
+            * integer, to specify the number of folds in a (Stratified)KFold,
+            * An object to be used as a cross-validation generator.
+            * An iterable yielding train, test splits.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+        Used to generate random chain orders.
+
+    Attributes
+    ----------
+    classes_ : list
+        A list of arrays of length len(estimators_) containing the
+        class labels for each estimator in the chain.
+
+    estimators_ : list
+        A list of clones of base_estimator.
+
+    order_ : list
+        The order of labels in the classifier chain.
+
+    References
+    ----------
+    Jesse Read, Bernhard Pfahringer, Geoff Holmes, Eibe Frank, "Classifier
+    Chains for Multi-label Classification", 2009.
+
+    """
+    def __init__(self, base_estimator, order=None, cv=None, random_state=None):
+        self.base_estimator = base_estimator
+        self.order = order
+        self.cv = cv
+        self.random_state = random_state
+
+    def fit(self, X, Y):
+        """Fit the model to data matrix X and targets Y.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+        Y : array-like, shape (n_samples, n_classes)
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, Y = check_X_y(X, Y,  multi_output=True, accept_sparse=True)
+
+        random_state = check_random_state(self.random_state)
+        check_array(X, accept_sparse=True)
+        self.order_ = self.order
+        if self.order_ is None:
+            self.order_ = np.array(range(Y.shape[1]))
+        elif isinstance(self.order_, str):
+            if self.order_ == 'random':
+                self.order_ = random_state.permutation(Y.shape[1])
+        elif sorted(self.order_) != list(range(Y.shape[1])):
+                raise ValueError("invalid order")
+
+        self.estimators_ = [clone(self.base_estimator)
+                            for _ in range(Y.shape[1])]
+
+        self.classes_ = []
+
+        if self.cv is None:
+            Y_pred_chain = Y[:, self.order_]
+            if sp.issparse(X):
+                X_aug = sp.hstack((X, Y_pred_chain), format='lil')
```

Comment:
I only expected that warning when inserting into a CSC/CSR, and I thought we don't insert in the cv is None case